### PR TITLE
Austen's Solution

### PR DESCRIPTION
**Pull request by austenmadden**
@jaybobo @sampriddy @sjreich

### CODE REVIEW

File: lib/tree.rb [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
+    @height = 1
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    if @age >= 100 then @alive = false end
+    if @alive == true
```
Comment (paradime) [inline lib/tree.rb:16]: This can be shortened to just @alive because if it's true, it will resolve the if block

File: lib/tree.rb [hunk 2]
```diff
@@ -1,38 +1,59 @@
...
 
   def add_apples
+    apples_produced = rand(100..200)
+    for i in 0..apples_produced
+      @apples.push(Apple.new("red", rand(1.5..3.5)))
+    end
   end
 
   def any_apples?
+    if @apples.count >= 1 then true else false end
```
Comment (paradime) [inline lib/tree.rb:30]: You can simplify this with just `@apples.count >= 1`

File: lib/tree.rb [hunk 3]
```diff
@@ -1,38 +1,59 @@
...
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+     apple = @apples.shift
```
Comment (paradime) [inline lib/tree.rb:35]: Ruby will return the last statement so you don't even need this assignment

File: spec/tree_spec.rb [hunk 4]
```diff
@@ -1,14 +1,77 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe 'AppleTree' do
```
Comment (paradime) [inline spec/tree_spec.rb:4]: You can remove the quotes here. If you do that, you can reference AppleTree anywhere with `described_class`

File: spec/tree_spec.rb [hunk 5]
```diff
@@ -1,14 +1,77 @@
...
+
+  tree = AppleTree.new
```
Comment (paradime) [inline spec/tree_spec.rb:6]: It is best practice to use either a let(:tree) ... here or declare the tree a `subject`

File: spec/tree_spec.rb [hunk 6]
```diff
@@ -1,14 +1,77 @@
...
+    expect(before_count < tree.apples.count).to eq true
+  end
+
+  it 'should lose apples when picked' do
+    before_count = tree.apples.count
+    tree.pick_an_apple!
+    expect(before_count > tree.apples.count).to eq true
   end
+
+  it 'should die' do
```
Comment (paradime) [inline spec/tree_spec.rb:36]: Morbid!

File: lib/tree.rb (hunk 1 above)
Comment (austenmadden) [inline lib/tree.rb:16]: So you mean something along the lines of 

``` Ruby
if @alive
  blah
  blah 
end
```

Comment (paradime) [inline lib/tree.rb:16]: Yes

File: spec/tree_spec.rb (hunk 6 above)
Comment (austenmadden) [inline spec/tree_spec.rb:36]: It gave me a good chuckle

**Pull request closed without merging**